src/main/java/com/amplitude/Amplitude.java

```diff
@@ -57,6 +61,17 @@ public static Amplitude getInstance(String instanceName) {
    */
   public void init(String key) {
     apiKey = key;
+    httpCall = new GeneralHttpCall(apiKey);
+  }
+
+  public void setHttpCallMode(Boolean isBatchMode) {
+    if (isBatchMode) {
+      httpCallMode = HttpCallMode.BATCH_HTTPCALL;
+      httpCall = new BatchHttpCall(apiKey);
+    } else {
+      httpCallMode = HttpCallMode.REGULAR_HTTPCALL;
+      httpCall = new GeneralHttpCall(apiKey);
```

Comment:
one thing I feel is the httpCall here doesn't really rely on the httpCallMode, and if I remove the httpCallMode, it won't really affect or break anything... like in init, it just initiate on with genera but not really caring about the httpCallMode. One thing we can consider is have a method createHttpCall or updateHttpCall, with a param HttpCallMode, inside it checks if there is update to call mode, then update httpCall as necessary, and use that as the only method that can update httpCall. So we don't have multiple methods or place to update httpCall.

Reply:
Yea, I realize the same thing. You are right we should check if the current HttpCallMode is changed or not. If it's not changed, we don't need to do anything.